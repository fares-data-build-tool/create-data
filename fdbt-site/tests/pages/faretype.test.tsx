import * as React from 'react';
import { shallow } from 'enzyme';
import FareType, { buildUuid } from '../../src/pages/fareType';

describe('pages', () => {
    const mockErrors = [{ errorMessage: 'Choose a fare type from the options', id: 'fare-type-error' }];

    describe('fareType', () => {
        it('should render correctly', () => {
            const tree = shallow(<FareType operator={' '} errors={[]} csrfToken="" />);
            expect(tree).toMatchSnapshot();
        });

        it('should render error messaging when errors are passed to the page', () => {
            const tree = shallow(<FareType operator={' '} errors={mockErrors} csrfToken="" />);
            expect(tree).toMatchSnapshot();
        });
    });

    describe('buildUUid', () => {
        it('should return a string starting with the NOC and then 8 characters of uuid', () => {
            const result = buildUuid('TEST');

            expect(result.substring(0, 4)).toBe('TEST');
            expect(result.length).toBe(12);
        });
    });
});
