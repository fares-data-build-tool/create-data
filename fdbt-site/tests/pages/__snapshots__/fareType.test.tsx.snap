// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages fareType should render correctly 1`] = `
<Component
  description="Fare Type selection page of the Create Fares Data Service"
  errors={Array []}
  title="Fare Type - Create Fares Data Service "
>
  <CsrfForm
    action="/api/fareType"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="fare-type-page-heading"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="fare-type-page-heading"
          >
            Select a fare type
          </h1>
        </legend>
        <span
          className="govuk-hint"
          id="fare-type-operator-hint"
        >
           
        </span>
        <FormElementWrapper
          errorClass="govuk-radios--error"
          errorId="fare-type-single"
          errors={Array []}
        >
          <FareTypeRadios
            otherFares={
              Array [
                Object {
                  "fareType": "multiOperator",
                  "label": "Multi-operator - a ticket that covers more than one operator",
                },
                Object {
                  "fareType": "schoolService",
                  "label": "School service - a ticket available to pupils in full-time education",
                },
              ]
            }
            standardFares={
              Array [
                Object {
                  "fareType": "single",
                  "label": "Single ticket - point to point",
                },
                Object {
                  "fareType": "period",
                  "label": "Period ticket (day, week, month and annual)",
                },
                Object {
                  "fareType": "return",
                  "label": "Return ticket - single service",
                },
                Object {
                  "fareType": "flatFare",
                  "label": "Flat fare ticket - single journey",
                },
              ]
            }
          />
        </FormElementWrapper>
      </fieldset>
    </div>
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;

exports[`pages fareType should render correctly for a scheme operator 1`] = `
<Component
  description="Fare Type selection page of the Create Fares Data Service"
  errors={Array []}
  title="Fare Type - Create Fares Data Service "
>
  <CsrfForm
    action="/api/fareType"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="fare-type-page-heading"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="fare-type-page-heading"
          >
            Select a fare type
          </h1>
        </legend>
        <span
          className="govuk-hint"
          id="fare-type-operator-hint"
        >
           
        </span>
        <FormElementWrapper
          errorClass="govuk-radios--error"
          errorId="fare-type-single"
          errors={Array []}
        >
          <FareTypeRadios
            otherFares={Array []}
            standardFares={
              Array [
                Object {
                  "fareType": "period",
                  "label": "Period geozone ticket (day, week, month and annual in a zone)",
                },
                Object {
                  "fareType": "flatFare",
                  "label": "Flat fare ticket - single journey",
                },
              ]
            }
          />
        </FormElementWrapper>
      </fieldset>
    </div>
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;

exports[`pages fareType should render error messaging when errors are passed to the page 1`] = `
<Component
  description="Fare Type selection page of the Create Fares Data Service"
  errors={
    Array [
      Object {
        "errorMessage": "Choose a fare type from the options",
        "id": "fare-type-single",
      },
    ]
  }
  title="Fare Type - Create Fares Data Service "
>
  <CsrfForm
    action="/api/fareType"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={
        Array [
          Object {
            "errorMessage": "Choose a fare type from the options",
            "id": "fare-type-single",
          },
        ]
      }
    />
    <div
      className="govuk-form-group govuk-form-group--error"
    >
      <fieldset
        aria-describedby="fare-type-page-heading"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="fare-type-page-heading"
          >
            Select a fare type
          </h1>
        </legend>
        <span
          className="govuk-hint"
          id="fare-type-operator-hint"
        >
           
        </span>
        <FormElementWrapper
          errorClass="govuk-radios--error"
          errorId="fare-type-single"
          errors={
            Array [
              Object {
                "errorMessage": "Choose a fare type from the options",
                "id": "fare-type-single",
              },
            ]
          }
        >
          <FareTypeRadios
            otherFares={
              Array [
                Object {
                  "fareType": "multiOperator",
                  "label": "Multi-operator - a ticket that covers more than one operator",
                },
                Object {
                  "fareType": "schoolService",
                  "label": "School service - a ticket available to pupils in full-time education",
                },
              ]
            }
            standardFares={
              Array [
                Object {
                  "fareType": "single",
                  "label": "Single ticket - point to point",
                },
                Object {
                  "fareType": "period",
                  "label": "Period ticket (day, week, month and annual)",
                },
                Object {
                  "fareType": "return",
                  "label": "Return ticket - single service",
                },
                Object {
                  "fareType": "flatFare",
                  "label": "Flat fare ticket - single journey",
                },
              ]
            }
          />
        </FormElementWrapper>
      </fieldset>
    </div>
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;
